spec/ref_arch_setup/bolt_helper_spec.rb

```diff
@@ -1,19 +1,24 @@
 require "spec_helper"
 
 describe RefArchSetup::BoltHelper do
-  let(:nodes)        { "local://localhost" }
-  let(:dir)          { "/tmp/ref_arch_setup" }
-  let(:cmd)          { "echo foo" }
-  let(:task)         { "ref_arch_setup::foo" }
-  let(:params)       { { "VAR1" => "1", "VAR2" => "2" } }
-  let(:params_str)   { "VAR1=1 VAR2=2" }
-  let(:source)       { "/tmp/foo" }
-  let(:destination)  { "/tmp/bar" }
+  let(:nodes)             { "local://localhost" }
+  let(:dir)               { "/tmp/ref_arch_setup" }
+  let(:cmd)               { "echo foo" }
+  let(:task)              { "ref_arch_setup::foo" }
+  let(:params)            { { "VAR1" => "1", "VAR2" => "2" } }
+  let(:params_str)        { "VAR1=1 VAR2=2" }
+  let(:bolt_user_opts)    { { "user" => "my_user", "password" => "my_password" } }
+  let(:bolt_user_string)  { "--user my_user --password my_password" }
+  let(:bolt_pkey_opts)    { { "private-key" => "private_key_path" } }
+  let(:bolt_pkey_string)  { "--private-key private_key_path" }
+  let(:source)            { "/tmp/foo" }
+  let(:destination)       { "/tmp/bar" }
 
   describe "make_dir" do
     before do
       @expected_command = "[ -d #{dir} ] ||"
       @expected_command << " mkdir -p #{dir}"
+      RefArchSetup::BoltHelper.bolt_options = {}
```

Comment:
This feels like it shouldn't need to happen.  Like the module should handle it somehow without us needing to do this.  Is that possible?

Reply:
This is because the methods in BoltHelper are implemented as static methods, meaning there is only the static singleton instance. If this were a class where we created a new instance, or a module we included in another class or module then it would be pretty easy. In this case, our unit tests are doing something we don't ever expect a user to do, which is changing the value after it has been initialized.